### PR TITLE
Use Rust FFI #12: Binary

### PR DESCRIPTION
**Pull request by tienvx**
Add binary file support with example.

This PR depends on https://github.com/pact-foundation/pact-php/pull/331. Please review it first.

### CODE REVIEW

File: example/binary/consumer/tests/Service/HttpClientServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        $request
+            ->setMethod('GET')
+            ->setPath('/image.jpg')
+            ->addHeader('Accept', 'image/jpeg');
+
+        $response = new ProviderResponse();
+        $response
+            ->setStatus(200)
+            ->addHeader('Content-Type', 'image/jpeg')
+            ->setBody(new Binary($imageContent, in_array(php_uname('m'), ['AMD64', 'arm64']) ? 'application/octet-stream' : 'image/jpeg'));
```
Comment (YOU54F) [inline example/binary/consumer/tests/Service/HttpClientServiceTest.php:29]: Related to https://github.com/pact-foundation/pact-reference/issues/305 I presume

**Review by YOU54F — APPROVED** [review]
Looks good dude, I do like the separation of the examples by function, as it's not clear for example from pact-js v3 folder

https://github.com/pact-foundation/pact-js/tree/master/examples/v3

as to the fact the to-do consumer is the xml project (i know it says elsewhere but there folders could be desrcription). food for thought as we begin to align project examples to had a common scenario and reference to then have a sample test in each language)

File: example/binary/consumer/tests/Service/HttpClientServiceTest.php (hunk 1 above)
Comment (tienvx) [reply example/binary/consumer/tests/Service/HttpClientServiceTest.php:29]: I think https://github.com/pact-foundation/pact-reference/issues/171 is more precise

Comment (YOU54F) [reply example/binary/consumer/tests/Service/HttpClientServiceTest.php:29]: ahh yes, spot on 🔍

**Pull request merged**